dandi/metadata.py

```diff
@@ -90,9 +90,86 @@ def get_metadata(path):
     return meta
 
 
+def _parse_iso8601(age):
+    """checking if age is proper iso8601, additional formatting"""
+    # allowing for comma instead of ., e.g. P1,5D
+    age = age.replace(",", ".")
+    pattern = (
+        "^P(?!$)(\\d+(?:\\.\\d+)?Y)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?W)?(\\d+(?:\\.\\d+)?D)?"
+        "(T(?=\\d)(\\d+(?:\\.\\d+)?H)?(\\d+(?:\\.\\d+)?M)?(\\d+(?:\\.\\d+)?S)?)?$"
+    )
+
+    matchstr = re.match(pattern, age, flags=re.I)
+    if matchstr:
+        age_rem = age.replace(matchstr.group(0), "")
+        if age_rem:
+            raise ValueError(
+                f"ISO 8601 expected, but {age} was received,"
+                f"no rules to parse {age_rem}"
+            )
+        age_frm = [matchstr.group(i) for i in range(1, 6) if matchstr.group(i)]
+        age_frm = ["P"] + age_frm
+        return age_frm
+    else:
+        raise ValueError(f"ISO 8601 expected, but {age} was received")
+
+
+def _parse_age_re(age, unit, tp="date"):
+    """finding parts that have <value> <unit> in various forms"""
+
+    if unit == "Y":
+        pat_un = "y(ear)?"
+    elif unit == "M" and tp == "date":
+        pat_un = "(month|mon|mo|m)"
+    elif unit == "W":
+        pat_un = "w(eek)?"
+    elif unit == "D":
+        pat_un = "d(ay)?"
+    elif unit == "H":
+        pat_un = "h(our)?"
+    elif unit == "M" and tp == "time":
+        pat_un = "(min|m(inute)?)"
+    elif unit == "S":
+        pat_un = "(sec|s(econd)?)"
+
+    pattern = rf"(\d+\.?\d*)\s*({pat_un}s?)"
+    matchstr = re.match(pattern, age, flags=re.I)
+    if matchstr is None:
+        # checking pattern with "unit" word
+        pattern_unit = rf"(\d+\.?\d*)\s*units?:?\s*({pat_un}s?)"
+        matchstr = re.match(pattern_unit, age, flags=re.I)
+        if matchstr is None:
+            return age, None
+
+    if "." in matchstr.group(1):
+        qty = float(matchstr.group(1))
+        if int(qty) == qty:
+            qty = int(qty)
```

Comment:
this code block of `if` isn't tests covered.  I've not read code to detail yet -- do you @djarecka see a quick test case to  add to tests to catch this one as well, just to be on a safe side? (code looks logical to me)

Reply:
added